linkerd/http-route/src/http/filter/redirect.rs

```diff
@@ -193,6 +193,84 @@ impl RedirectRequest {
     }
 }
 
+#[cfg(feature = "proto")]
+pub mod proto {
+    use super::*;
+    use linkerd2_proxy_api::{http_route as api, http_types};
+
+    #[derive(Debug, thiserror::Error)]
+    pub enum InvalidRequestRedirect {
+        #[error("invalid location scheme: {0}")]
+        Scheme(#[from] http_types::InvalidScheme),
+
+        #[error("invalid HTTP status code: {0}")]
+        Status(#[from] http::status::InvalidStatusCode),
+
+        #[error("invalid HTTP authority: {0}")]
+        Authority(#[from] http::uri::InvalidUri),
+
+        #[error("invalid HTTP status code: {0}")]
+        StatusNonU16(u32),
+
+        #[error("invalid port number: {0}")]
+        Port(u32),
+
+        #[error("{0}")]
+        Value(#[from] http::header::InvalidHeaderValue),
+    }
+
+    // === impl RedirectRequest ===
+
+    impl TryFrom<api::RequestRedirect> for RedirectRequest {
+        type Error = InvalidRequestRedirect;
+
+        fn try_from(rr: api::RequestRedirect) -> Result<Self, Self::Error> {
+            let scheme = rr.scheme.map(TryInto::try_into).transpose()?;
+
+            let authority = {
+                if rr.port > (u16::MAX as u32) {
+                    return Err(InvalidRequestRedirect::Port(rr.port));
+                }
+                match (rr.host, (rr.port as u16).try_into().ok()) {
+                    (h, p) if h.is_empty() => p.map(AuthorityOverride::Port),
+                    (h, Some(p)) => {
+                        let a = format!("{}:{}", h, p).try_into()?;
+                        Some(AuthorityOverride::Exact(a))
+                    }
+                    (h, None) => {
+                        let a = h.try_into()?;
+                        Some(AuthorityOverride::Host(a))
+                    }
+                }
+            };
+
+            let path = rr.path.and_then(|p| p.replace).map(|p| match p {
+                api::path_modifier::Replace::Full(path) => {
+                    // TODO ensure path is valid.
```

Comment:
assuming these TODOs are going to be addressed in a subsequent PR?

Reply:
See TODOs in PR description. I'm not sure it's required.